### PR TITLE
batched sentiment inference and proper scoring

### PR DESCRIPTION
**Pull request by jtwinrelevanceai**

### CODE REVIEW

File: relevanceai/operations_new/sentiment/transform.py [hunk 1]
```diff
@@ -98,60 +98,53 @@ def label_mapping(self):
...
-            new_ind_max = np.argmax([l["score"] for l in new_labels])
-            new_max_score = new_labels[new_ind_max]["score"]
-            new_sentiment = new_labels[new_ind_max]["label"]
-            new_sentiment = self.label_mapping.get(new_sentiment, new_sentiment)
-            overall_sentiment = self._calculate_overall_sentiment(
-                new_max_score, new_sentiment
-            )
+        labels = self.classifier(texts, truncation=True, max_length=512)
+        scores = np.array([[l["score"] for l in output] for output in labels])
+        e_x = scores[:, 0] * -1 + scores[:, 1] * 0 + scores[:, 2] * 1
```
Comment (boba-and-beer) [inline relevanceai/operations_new/sentiment/transform.py:110]: This assumes `negative` is always the first thing that is returned. Can you check this is true for all models? I don't think this is the case

**Review by boba-and-beer — CHANGES_REQUESTED** [review]
I don't think the logic is right

File: relevanceai/operations_new/sentiment/transform.py (hunk 1 above)
Comment (jtwinrelevanceai) [reply relevanceai/operations_new/sentiment/transform.py:110]: we have a hardcoded label mapping tho

File: relevanceai/operations_new/sentiment/transform.py [hunk 2]
```diff
@@ -96,62 +97,69 @@ def _get_label_mapping(self, task: str):
     def label_mapping(self):
         return {"LABEL_0": "negative", "LABEL_1": "neutral", "LABEL_2": "positive"}
 
+    def _get_scores(
+        self, cls: str, logits: List[List[Dict[str, float]]]
+    ) -> List[float]:
+        return [
+            list(filter(lambda label: label["label"] == cls, logit))[0]["score"]
```
Comment (boba-and-beer) [inline relevanceai/operations_new/sentiment/transform.py:104]: I thikn we need to do some string normalization here

**Review by boba-and-beer — APPROVED** [review]

File: relevanceai/operations_new/sentiment/transform.py [hunk 3]
```diff
@@ -100,62 +92,131 @@ def _get_label_mapping(self, task: str):
...
-        if overall_sentiment == 0:
-            overall_sentiment = 1e-5
-        if not highlight:
-            return {
-                "sentiment": sentiment,
-                "overall_sentiment_score": overall_sentiment,
+        logits = self.classifier(texts, truncation=True, max_length=512)
+        scores = pd.DataFrame(
+            {
+                "negative": self._get_scores(cls="LABEL_0", logits=logits),
```
Comment (boba-and-beer) [inline relevanceai/operations_new/sentiment/transform.py:112]: This would only work on 1 of the sentiment models

**Review by boba-and-beer — APPROVED** [review]

**Review by boba-and-beer — APPROVED** [review]
